features/deep-linking/conversation-links.ts

```diff
@@ -0,0 +1,41 @@
+import { IXmtpConversationId, IXmtpInboxId } from "@/features/xmtp/xmtp.types"
+import { findConversationByInboxIds } from "@/features/conversation/utils/find-conversations-by-inbox-ids"
+import { useMultiInboxStore } from "@/features/authentication/multi-inbox.store"
+import { logger } from "@/utils/logger"
+
+/**
+ * Check if a conversation exists with the given inboxId
+ * @param inboxId The inbox ID to check
+ * @returns Promise with { exists, conversationId } - where conversationId is set if exists is true
+ */
+export async function checkConversationExists(inboxId: IXmtpInboxId): Promise<{ exists: boolean, conversationId?: IXmtpConversationId }> {
```

Comment:
I also don't like the explicty return. I think we should always implicit, should be in cursorrules weird it did this